src/Features/CSharp/Portable/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProvider.cs

```diff
@@ -45,7 +45,7 @@ protected override (string displayText, string suffix, string insertionText) Get
         protected override async Task<SyntaxContext> CreateContextAsync(
             Document document, int position, CancellationToken cancellationToken)
         {
-            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.ReuseExistingSpeculativeModelAsync(position, cancellationToken).ConfigureAwait(false);
```

Comment:
Yes.  I think what we need to do instead is have creating the context happen in the base type, and have that context be passed around.  The context will eagerly create the speculative model (and name it accordingly) and pass it directly to abstrat methods.  This will make it clearer that this is the model you should be using.